kernel/shell.c

```diff
@@ -0,0 +1,325 @@
+#include <dirent.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+#include <myst/chars.h>
+#include <myst/debugmalloc.h>
+#include <myst/fdtable.h>
+#include <myst/id.h>
+#include <myst/kernel.h>
+#include <myst/mmanutils.h>
+#include <myst/panic.h>
+#include <myst/printf.h>
+#include <myst/strings.h>
+#include <myst/syscall.h>
+
+#if !defined(MYST_RELEASE)
+
+#define COLOR_LIGHT_BLUE "\e[94m"
+#define COLOR_LIGHT_GREEN "\e[92m"
+#define COLOR_RESET "\e[0m"
+
+__attribute__((__unused__)) static void _dump_args(size_t argc, char** argv)
+{
+    printf("argc=%zu\n", argc);
+
+    for (size_t i = 0; i < argc; i++)
+        printf("argv[%zu]=%s\n", i, argv[i]);
+}
+
+typedef struct help
+{
+    const char* cmd;
+    const char* msg;
+} help_t;
+
+static help_t _help[] = {
+    {"help", "print this message"},
+    {"ls", "list directory contents"},
+    {"cd", "change the current directory"},
+    {"pwd", "print the current directory"},
+    {"mem", "print memory statistics"},
+    {"cont", "leave shell and ocntinue execution"},
+    {"fds", "list open file descriptors"},
+    {"id", "print the current UID and GID"},
+    {"maxthreads", "list open file descriptors"},
```

Comment:
The description for maxthreads is wrong.

Reply:
It would be nice to report the current number of running threads / processes.

Reply:
Fixed and added ``numthreads`` command.